easydock/database.py

```diff
@@ -184,7 +184,34 @@ def restore_setup_from_db(db_fname, tmpdir=None):
 
     return d, tmpfiles
 
+import signal
+import functools
 
+#https://stackoverflow.com/questions/75928586/how-to-stop-the-execution-of-a-function-in-python-after-a-certain-time/75928879#75928879
+def timeout(seconds=5, default=None):
```

Comment:
I suggest to move this function to auxiliary module, since it is general and may be used from other modules in future.